tests/api.c

```diff
@@ -36699,6 +36699,60 @@ static void test_wolfSSL_EVP_PKEY_assign_DH(void)
     printf(resultFmt, passed);
 #endif
 }
+static void test_wolfSSL_EVP_PKEY_param_check(void)
+{
+#if defined(OPENSSL_ALL) || defined(WOLFSSL_QT)
+#if !defined(NO_DH) && !defined(NO_FILESYSTEM)
+#if !defined(HAVE_FIPS) || (defined(HAVE_FIPS_VERSION) \
+                                     && (HAVE_FIPS_VERSION>2))
+    DH       *dh    = NULL;
+    DH       *setDh = NULL;
+    EVP_PKEY *pkey  = NULL;
+    EVP_PKEY_CTX*   ctx = NULL;
+    
+    FILE* f = NULL;
+    unsigned char buf[4096];
```

Comment:
Does this array need to be 4096?  Looks like we read dh2048.der into it, but that file is only 268 bytes in size.

Reply:
That's true. Changed to 512 bytes for buffer.